replica/table.cc

```diff
@@ -3688,13 +3693,16 @@ future<> table::cleanup_compaction_groups(database& db, db::system_keyspace& sys
 
 future<> table::cleanup_tablet(database& db, db::system_keyspace& sys_ks, locator::tablet_id tid) {
     auto holder = async_gate().hold();
-    auto& sg = storage_group_for_id(tid.value());
 
-    co_await clear_inactive_reads_for_tablet(db, sg);
-    // compaction_group::stop takes care of flushing.
-    co_await stop_compaction_groups(sg);
-    co_await cleanup_compaction_groups(db, sys_ks, tid, sg);
-    _sg_manager->remove_storage_group(tid.value());
+    if (auto* sg = _sg_manager->maybe_storage_group_for_id(_schema, tid.value())) {
+        co_await clear_inactive_reads_for_tablet(db, *sg);
+        // compaction_group::stop takes care of flushing.
+        co_await stop_compaction_groups(*sg);
+        co_await cleanup_compaction_groups(db, sys_ks, tid, *sg);
+        _sg_manager->remove_storage_group(tid.value());
+        co_return;
+    }
+    tlogger.warn("Storage group for tablet {} is deallocated. Ignore cleanup.", tid);
```

Comment:
To reduce the code churn, you could reverse the logic order, on these line:
```
auto sgp = _sg_manager->maybe_storage_group_for_id(_schema, tid.value());
if (!sgp) {
    tlogger.warn("Storage group for tablet {} is deallocated. Ignore cleanup.", tid);
    co_return;
}

auto& sg = *sgp;
...
```

Reply:
It's a bit odd to put the error-free path in a block that co_return's and then have a warning in the "default" path (which is actually the benign error path)